docs/src/references/mqtt-topic-guidelines.md

```diff
@@ -93,13 +95,19 @@ Having them in the topics is desired, as that enables easy filtering of messages
    * All measurements from the tedge device only, excluding the ones from other services and child devices
    * All measurements from all connected child devices
    * All measurements from everything (the tedge device itself, its services and child devices and their services)
+   * All messages from a subset of child devices based on some filtering criteria (e.g: device-type, firmware version etc)
 1. Service ids must be namespaced under each device as it is highly likely that
    the same type of service may have the same name on multiple/all devices.
    It would be really difficult to maintain uniqueness in service name across an large fleet of devices.
 1. Child device ids also must be namespaced under their direct parent
    so that conflicts can be avoided even if another parent device has child devices with the same id.
    It is okay to expect all devices connected to a device to have unique ids,
    but expecting those to be unique across an entire fleet could be far-fetched.
+1. If all child devices in a fleet can not guarantee uniqueness in their IDs,
+   thin-edge must provide a "registration mechanism" for them to get their own unique IDs,
+   which they can use for all their HTTP/MQTT communications.
```

Comment:
We expect all the child devices connected to a parent to have unique names, at least under the namespace of that parent. This is the basic requirement that we can easily enforce. The issue is in enforcing that across multiple parent devices. What I'm not sure is whether it is okay to expect that even children connected to different parents would not have the same name.

Here is the kind of deployment that I'm worried about:

```
              tedge
                |
             child-a
            /         \
child-b             child-c
                                \
                                 child-x
                                /
                    child-a
                  /
       child-b
```
... where even a parent-child ID combination is not unique under that tedge namespace, even-though all parents have uniquely named children in isolation.

Reply:
> The issue is in enforcing the same across multiple parent devices. What I'm not sure is whether it is okay to expect that even children connected to different parents would not have the same name.

No, it's not okay to expect such a strong requirement. This would prevent basic settings with child-devices of the same type (same hardware, firmware and software).

But why do you see a need to enforce that? Knowing that you propose topics with a hierarchy a child devices.

Reply:
Oh, no I **don't** want to capture the entire device hierarchy in the topic of each and every message. Of course, that solves this uniqueness problem, but lead to a lot of redundant data in each and every message. So, my goal was to define a scheme without complete hierarchy in the topic. I also wanted to limit the expected "hierarchical knowledge" of a device to just itself and its immediate parent only, and **not** its grand-parents and above.